pkg/operator/ceph/csi/controller.go

```diff
@@ -203,28 +214,40 @@ func (r *ReconcileCSI) reconcile(request reconcile.Request) (reconcile.Result, e
-			if err != nil {
-				// This avoids a requeue with exponential backoff and allows the controller to reconcile
-				// more quickly when the cluster is ready.
-				if errors.Is(err, opcontroller.ClusterInfoNoClusterNoSecret) {
-					logger.Infof("cluster info for cluster %q is not ready yet, will retry in %s, proceeding with ready clusters", cluster.Name, opcontroller.WaitForRequeueIfCephClusterNotReady.RequeueAfter.String())
-					reconcileResult = opcontroller.WaitForRequeueIfCephClusterNotReady
-					continue
-				}
-				return opcontroller.ImmediateRetryResult, errors.Wrapf(err, "failed to load cluster info for cluster %q", cluster.Name)
+		// Load cluster info for later use in updating the ceph-csi configmap
+		clusterInfo, _, _, err := opcontroller.LoadClusterInfo(r.context, r.opManagerContext, cluster.Namespace, &cephClusters.Items[i].Spec)
+		if err != nil {
+			// This avoids a requeue with exponential backoff and allows the controller to reconcile
+			// more quickly when the cluster is ready.
+			if errors.Is(err, opcontroller.ClusterInfoNoClusterNoSecret) {
+				logger.Infof("cluster info for cluster %q is not ready yet, will retry in %s, proceeding with ready clusters", cluster.Name, opcontroller.WaitForRequeueIfCephClusterNotReady.RequeueAfter.String())
+				reconcileResult = opcontroller.WaitForRequeueIfCephClusterNotReady
+				continue
 			}
-			clusterInfo.OwnerInfo = k8sutil.NewOwnerInfo(&cephClusters.Items[i], r.scheme)
-			logger.Debugf("cluster %q is running on multus or CSI_ENABLE_HOST_NETWORK is false, deploying the ceph-csi plugin holder", cluster.Name)
+			return opcontroller.ImmediateRetryResult, errors.Wrapf(err, "failed to load cluster info for cluster %q", cluster.Name)
+		}
+		clusterInfo.OwnerInfo = k8sutil.NewOwnerInfo(&cephClusters.Items[i], r.scheme)
+
+		// is holder enabled for this cluster?
+		thisHolderEnabled := (!csiHostNetworkEnabled || cluster.Spec.Network.IsMultus()) && !csiDisableHolders
 
+		// Do we have a multus cluster or csi host network disabled?
+		// If so deploy the plugin holder with the fsid attached
+		if thisHolderEnabled {
+			logger.Debugf("cluster %q: deploying the ceph-csi plugin holder", cluster.Name)
 			r.clustersWithHolder = append(r.clustersWithHolder, ClusterDetail{cluster: &cephClusters.Items[i], clusterInfo: clusterInfo})
+
+			// holder pods are enabled globally if any cluster needs a holder pod
+			holderEnabled = true
 		} else {
-			logger.Debugf("not a multus cluster %q or CSI_ENABLE_HOST_NETWORK is true, not deploying the ceph-csi plugin holder", request.NamespacedName)
+			logger.Debugf("cluster %q: not deploying the ceph-csi plugin holder", request.NamespacedName)
+		}
+
+		// if holder pods were disabled, the controller needs to update the configmap for each
+		// cephcluster to remove the net namespace file path
+		err = SaveCSIDriverOptions(r.context.Clientset, cluster.Namespace, clusterInfo)
```

Comment:
SaveClusterConfig will not remove the netnamespace from the configmap (from my code reading i could be wrong but we need to check on this)

we have multiple functions acting the same configmap in different ways which can easily introduce bugs, we need to refractor it (not as part of this PR but later for sure)

Reply:
I added `updateNetNamespaces()` to that function so I could use it here.

Reply:
https://github.com/rook/rook/pull/13890/files#r1521688482